src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -74,24 +106,63 @@ class MoneyRequestAmountPage extends React.Component {
     }
 
     componentDidMount() {
-        this.focusTextInput();
+        if (this.isEditing) {
```

Comment:
Hmm I wonder if we should handle these shouldReset / isEditing in a different way that is more DRY

Are these to handle new money request drafts?

Reply:
I was thinking to create a custom hook as the reset logic is being used in the amount, participant, confirmation, and description page, but both the amount and description page is still class component.

> Are these to handle new money request drafts?

The reset logic is to prevent the user from doing a money request of type A with money request data of type B. For example, doing a send request with saved split request data. This is possible by directly accessing the URL (change from /send/new/confirmation to /split/new/confirmation). More context here https://github.com/Expensify/App/pull/17964#issuecomment-1584422120